app/src/main/kotlin/com/adesso/movee/base/BaseFragmentDecorator.kt

```diff
@@ -1,56 +1,30 @@
 package com.adesso.movee.base
 
-import androidx.annotation.IdRes
 import androidx.core.net.toUri
 import androidx.databinding.ViewDataBinding
 import androidx.fragment.app.Fragment
-import androidx.fragment.app.activityViewModels
-import androidx.fragment.app.viewModels
-import androidx.lifecycle.ViewModel
 import androidx.lifecycle.ViewModelProvider
 import androidx.navigation.fragment.FragmentNavigator
 import androidx.navigation.fragment.FragmentNavigatorExtras
 import androidx.navigation.fragment.findNavController
-import androidx.navigation.navGraphViewModels
 import com.adesso.movee.BR
 import com.adesso.movee.internal.extension.observeNonNull
 import com.adesso.movee.internal.extension.showPopup
 import com.adesso.movee.internal.util.functional.lazyThreadSafetyNone
 import com.adesso.movee.navigation.NavigationCommand
 import com.adesso.movee.scene.main.MainActivity
 import com.google.android.material.snackbar.Snackbar
-import java.lang.reflect.ParameterizedType
 
 class BaseFragmentDecorator<VM : BaseAndroidViewModel, B : ViewDataBinding>(
     val fragment: Fragment,
     val binder: B,
-    val viewModelFactory: ViewModelProvider.Factory
+    val vMClass: Class<VM>
 ) {
-    @Suppress("UNCHECKED_CAST")
-    val viewModel by lazyThreadSafetyNone {
-        val persistentViewModelClass = (fragment.javaClass.genericSuperclass as ParameterizedType)
```

Comment:
Buradaki yapıyı koruyarak ek view model Class<> alımını engelleyebilirsin. Tüm activity ve fragmentlerda da Class<VM> almana gerek kalmamış olur.

`    val viewModel by lazyThreadSafetyNone {
        val persistentViewModelClass = (fragment.javaClass.genericSuperclass as ParameterizedType)
            .actualTypeArguments[0] as Class<VM>
        return@lazyThreadSafetyNone ViewModelProvider(fragment)
            .get(persistentViewModelClass)
    }`

Not: Benzer Hilt geçişini @mypat bir projede tamamlamıştı, yukarıdaki kod bloğu oradan bir örnek.

Reply:
Güncelledim, tekrar kontrol edebilir misin?